Mathlib/Algebra/Order/Antidiag/Pi.lean

```diff
@@ -74,5 +85,150 @@ where
 @[simp] lemma mem_finAntidiagonal {d : ℕ} {f : Fin d → μ} :
     f ∈ finAntidiagonal d n ↔ ∑ i, f i = n := (finAntidiagonal.aux d n).prop f
 
+/-!
+### `ι → μ`
+
+In this section, we transfer the antidiagonals in `Fin s.card → μ` to antidiagonals in `ι → s` by
+choosing an identification `s ≃ Fin s.card` and proving the end result does not depend on that
+choice.
+-/
+
+/-- The finset of functions `ι → μ` with support contained in `s` and sum `n`. -/
+def piAntidiag (s : Finset ι) (n : μ) : Finset (ι → μ) := by
+  refine (Fintype.truncEquivFinOfCardEq $ Fintype.card_coe s).lift
+    (fun e ↦ (finAntidiagonal s.card n).map ⟨fun f i ↦ if hi : i ∈ s then f (e ⟨i, hi⟩) else 0, ?_⟩)
+    fun e₁ e₂ ↦ ?_
+  · rintro f g hfg
+    ext i
+    simpa using congr_fun hfg (e.symm i)
+  · ext f
+    simp only [mem_map, mem_finAntidiagonal]
+    refine Equiv.exists_congr ((e₁.symm.trans e₂).arrowCongr $ .refl _) fun g ↦ ?_
+    have := Fintype.sum_equiv (e₂.symm.trans e₁) _ g fun _ ↦ rfl
+    aesop
+
+variable {s : Finset ι} {n : μ} {f : ι → μ}
+
+@[simp] lemma mem_piAntidiag : f ∈ piAntidiag s n ↔ s.sum f = n ∧ ∀ i, f i ≠ 0 → i ∈ s := by
+  rw [piAntidiag]
+  induction' Fintype.truncEquivFinOfCardEq (Fintype.card_coe s) using Trunc.ind with e
+  simp only [Trunc.lift_mk, mem_map, mem_finAntidiagonal, Embedding.coeFn_mk]
+  constructor
+  · rintro ⟨f, ⟨hf, rfl⟩, rfl⟩
+    rw [sum_dite_of_true fun _ ↦ id]
+    exact ⟨Fintype.sum_equiv e _ _ (by simp), by simp (config := { contextual := true })⟩
+  · rintro ⟨rfl, hf⟩
+    refine ⟨f ∘ (↑) ∘ e.symm, ?_, by ext i; have := not_imp_comm.1 (hf i); aesop⟩
+    rw [← sum_attach s]
+    exact Fintype.sum_equiv e.symm _ _ (by simp)
+
+@[simp] lemma piAntidiag_empty_zero : piAntidiag (∅ : Finset ι) (0 : μ) = {0} := by
+  ext; simp [Fintype.sum_eq_zero_iff_of_nonneg, funext_iff, not_imp_comm, ← forall_and]
+
+@[simp] lemma piAntidiag_empty_of_ne_zero (hn : n ≠ 0) : piAntidiag (∅ : Finset ι) n = ∅ :=
+  eq_empty_of_forall_not_mem fun x ↦ by simp [@eq_comm _ 0, hn.symm]
+
+lemma piAntidiag_empty (n : μ) : piAntidiag (∅ : Finset ι) n = if n = 0 then {0} else ∅ := by
```

Comment:
Is there any of all these lemma that might/should be marked `aesop`?

Reply:
Highly unclear to me. I would rather not for now, out of lack of application (and since adding aesop rules incurs a cost on every future aesop call). That's easily fixable in a later PR